qadence/backends/braket/convert_ops.py

```diff
@@ -81,13 +81,19 @@ def BraketOperation(block: PrimitiveBlock) -> Instruction:
         return two_qubit[operation](block.qubit_support[0], block.qubit_support[1])
 
     elif operation in two_qubit_parametrized:
-        (expr,) = block.parameters.expressions()  # type: ignore [attr-defined]
-        angle_value = evaluate(expr)
-        return two_qubit_parametrized[operation](
-            control=block.qubit_support[0],
-            target=block.qubit_support[1],
-            angle=angle_value,
-        )
+        ((uuid, expr),) = block.parameters.items()  # type: ignore [attr-defined]
+        if expr.is_number:
+            return two_qubit_parametrized[operation](
+                control=block.qubit_support[0],
+                target=block.qubit_support[1],
+                angle=evaluate(expr),
+            )
+        else:
+            return two_qubit_parametrized[operation](
+                control=block.qubit_support[0],
+                target=block.qubit_support[1],
+                angle=FreeParameter(uuid),
+            )
```

Comment:
I do not really understand what is happening here. If the expression is an actual `sympy.Expr` and you do not evaluate it, you'll have a bug, I think.

Reply:
look at how the single qubit parametric gate is done, we check if the expr is fixed (i.e., is_number). if not, its a a expression with trainable/non-trainable (but not fixed) parameters